src/actions/guilds/bans/GUILD_BAN_ADD.ts

```diff
@@ -0,0 +1,22 @@
+import { Action } from '../../../lib/structures/Action';
+import { extender } from '../../../util/Extender';
+
+import type { GuildBanAddDispatch } from '@klasa/ws';
+import type { Ban } from '../../../client/caching/structures/guilds/Ban';
+
+export default class CoreAction extends Action {
+
+	public check(): null {
+		return null;
+	}
+
+	public build(data: GuildBanAddDispatch): Ban | null {
+		const guild = this.client.guilds.get(data.d.guild_id);
+		return guild ? new (extender.get('Ban'))(this.client, data.d, guild) : null;
+	}
+
+	public cache(data: Ban): void {
+		data.guild.bans.set(data.id, data);
```

Comment:
We haven't set up directives of what exactly to do with those bans. Cache always? Make it a client option? The latter feels the best way to do it but it's out of this PR's scope.

Same thing for `channel.messages` getting populated non-stop at the current state, limited collections aren't a thing in PB yet.

Reply:
@bdistin what are your thoughts on this?

Reply:
Actually, since we'll eventually add customizable cache limits, we'll soon be able to do `cache: { bans: 0 }` or something and the set will short-circuit to returning itself.

But that's another PR.